Sparkbox Sprint 54

rkelbel opened the pull request:
# READY FOR REVIEW

## Summary
- #1007 
- #1001 
- #1004 
- #991 
- #1010 
- #1009 

### TASK LIST:
- [x] Add compiled CSS & JS
- [x] Run BackstopJS visual regression tests
- [x] Ensure Code Climate passes against the `develop` branch. Link here: https://codeclimate.com/github/SU-HSDO/suhumsci/pull/1008
- [x] Assign to Mike for review.

## Need Review By (Date)
asap

## Urgency
high

## Steps to Test
1. `npm run test` should pass in the root folder
2. If you are interested in the testing steps for associated PRs, please take a look at the individual PR's description and "Steps to Test" section.

## PR Checklist
- [PR Checklist](https://gist.github.com/sherakama/0ba17601381e3adbe0cad566ad4d80a5)
- [Sparkbox PR Checklist](../docs/SparkboxPRChecklist.md)

Commit:
Feat(STN-855): Lead Font Animation (#1001)

* feat: add hs-font-lead to array of classesToAnimate

- updated conditional that determines whether or not the loop should run
  based on the animationEnhancements length.

* feat: added lead font keyframes

* feat: added mixin hb-font-lead-animation

- moved reducedMotionPrefered into mixin
- moved hasAnimationEnhancements into mixin

* feat: included hb-font-lead-animation

- when .hs-font-lead is a sibling of .hs-font-splash and on Traditional
  theme the animation is shorter and delayed

* fix: added in missing translateY parameter

Commit:
Feat(STN-896): add js linting (#991)

* feat: create new eslintrc file airbnb-base

Originally this application was setup to simply extend the core Drupal
eslint rules which don't really make sense here. Utilizing airbnb-base
over airbnb proper was due to our lack of needs around react.

Additionally, some rules were added to account for scenarios that could
ultimately lead to a regression in application functionality.

* chore: remove old eslintrc that is using the drupal eslintrc as this
file seems to be unused and doesn't really make sense.

* chore: pull in eslint plugin and fix lint issues with webpack config

* chore: add lint:js script

- update webpack
- update webpack-cli
- add eslint/eslint-config-airbnb-base
- add eslint-webpack-plugin/eslint-plugin-import

* chore: compile js

* fix: adding root eslintrc back in to see if build passes

* fix: added extend back in to eslint

* feat: add root to theme config

* refactor: sync page-scroll up with sprint branch...looks like it was
missed in the rebase

* chore: add lint fix script

* refactor: in compliance with eslint refactoring iterator with an array
method

* chore: add eslint rule exclusion for unnamed functions

Commit:
Lead font experimental features toggle (#1004)

* refactor: Lead font is now depenedent upon the Experimental Features
toggle instead of Animations toggle.

- added conditional to enable using Experimental Features toggle for
  additional animations in the future

* chore: compile js

* chore: compile css

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1008-zyukwtvw8mx08mem3vf9vtj3kz2shws0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/619d6a55de4970d98be77a76

Commit:
fix: Identify & Remove Console Errors for Font Aweseome  (#1007)

* fix: utilizing decanter variable $fa-font-path we're adding in the path
to the font awesome module to prevent unnecessary console logs output by
decanter.

* style: add spacing for variable

* chore: compile css

* chore: add comment for Decanter override

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1008-zyukwtvw8mx08mem3vf9vtj3kz2shws0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/619d6a55de4970d98be77a76

Commit:
Fix(STN-619): Remove the white space below universal footer (#1010)

* fix(STN-619): override style adding the white pixel bar

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1008-zyukwtvw8mx08mem3vf9vtj3kz2shws0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/619d6a55de4970d98be77a76

Commit:
Fix(STN-906): Bigger text sizes in collections (#1009)

* fix(STN-906): collection font sizes to match rows

pookmish commented:
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr1008-zyukwtvw8mx08mem3vf9vtj3kz2shws0.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/619d6a55de4970d98be77a76
